stdlib/public/Concurrency/Actor.cpp

```diff
@@ -1642,15 +1647,28 @@ void ProcessOutOfLineJob::process(Job *job) {
   DefaultActorImpl *actor = self->Actor;
 
   delete self;
-
-  // Balances with the swift_release in defaultActorDrain()
-  swift_retain(actor);
   return defaultActorDrain(actor); // 'return' forces tail call
 }
 
 #endif /* !SWIFT_CONCURRENCY_ACTORS_AS_LOCKS */
 
 void DefaultActorImpl::destroy() {
+  HeapObject *object = asAbstract(this);
+  size_t retainCount = swift_retainCount(object);
```

Comment:
@kubamracek, I need a bit of your help here.

This change causes linker error when running in `test/embedded/concurrency-actors.swift`:

```text
ld: Undefined symbols:
  _swift_retainCount, referenced from:
      _swift_defaultActor_destroy in libswift_Concurrency.a[4](Actor.cpp.o)
```

What is the correct way to fix this test? Which library contains _swift_retainCount for embedded configuration?

Reply:
Looks like `swift_retainCount()` must be handled in `SILLinker::linkEmbeddedRuntimeFromStdlib()`. This can be achieved by handling `swift_retainCount()` ad-hoc (not affecting other 3 use cases of the `RuntimeFunctions.def`) or by adding `swift_retainCount()` to `RuntimeFunctions.def` (for consistency). @kubamracek, @eeckstein, which one do you think is a better approach?

Reply:
I think your change is fine